staging/src/k8s.io/apiserver/pkg/apis/config/types.go

```diff
@@ -96,4 +96,14 @@ type KMSConfiguration struct {
 	// Timeout for gRPC calls to kms-plugin (ex. 5s). The default is 3 seconds.
 	// +optional
 	Timeout *metav1.Duration
+	// cacheHealthyTTL is the duration of time, after the last successful health check, during which
+	// kube-apiserver will not recheck the health status of the kms-plugin and will continue to
+	// return healthy status.
+	// +optional
+	CacheHealthyTTL *metav1.Duration
+	// cacheUnHealthyTTL is the duration of time, after the last failed health check, during which
+	// kube-apiserver will not recheck the health status of the kms-plugin and will continue to
+	// return failed status.
+	// +optional
+	CacheUnHealthyTTL *metav1.Duration
```

Comment:
It's unclear why this TTL exists.
Do healthz checks get spammed when they fail? Can we rely on the natural external healthz retry period instead?

Reply:
The reason why CacheUnHealthyTTL is proposed (in addition to CacheHealthyTTL) is that users may want to use different frequency of checks depending on the health status of the plugin.
For example, while plugin is coming-up (and responding !OK) we may want to re-check the status more aggressively, so that the master may be placed online quicker (ex. when a master is behind a load-balancer).
However, when plugin responds OK, we may want to back-off to reduce the cost of operating the service and load on the KMS.